pretyping/pretyping.ml

```diff
@@ -186,9 +186,23 @@ type inference_flags = {
 let frozen_holes (sigma, sigma') =
   (); fun ev -> Evar.Map.mem ev (Evd.undefined_map sigma)
 
+let rec advance sigma evk =
```

Comment:
Could you add a comment explaining what this function does? It is not obvious for non-experts like me, and will ease maintenance as well.